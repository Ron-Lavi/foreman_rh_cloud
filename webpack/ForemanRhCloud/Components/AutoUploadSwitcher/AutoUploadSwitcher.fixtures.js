export const handleToggleResponse = { data: { autoUploadEnabled: true } };
