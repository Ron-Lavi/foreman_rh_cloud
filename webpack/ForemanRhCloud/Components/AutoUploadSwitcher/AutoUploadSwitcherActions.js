import API from 'foremanReact/API';
import {
  AUTO_UPLOAD_TOGGLE,
  AUTO_UPLOAD_TOGGLE_ERROR,
} from './AutoUploadSwitcherConstants';

export const handleToggle = currentAutoUploadEnabled => async dispatch => {
  const toggledAutoUploadEnabled = !currentAutoUploadEnabled;
  try {
    const {
      data: { autoUploadEnabled },
    } = await API.post('auto_upload', { value: toggledAutoUploadEnabled });
    dispatch({
      type: AUTO_UPLOAD_TOGGLE,
      payload: {
        autoUploadEnabled,
      },
    });
  } catch (error) {
    dispatch({
      type: AUTO_UPLOAD_TOGGLE_ERROR,
      payload: {
        error: error.message,
      },
    });
  }
};
