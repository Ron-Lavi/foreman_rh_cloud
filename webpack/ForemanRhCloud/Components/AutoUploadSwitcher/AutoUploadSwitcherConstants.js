export const AUTO_UPLOAD_TOGGLE = 'AUTO_UPLOAD_TOGGLE';
export const AUTO_UPLOAD_TOGGLE_ERROR = 'AUTO_UPLOAD_TOGGLE_ERROR';
