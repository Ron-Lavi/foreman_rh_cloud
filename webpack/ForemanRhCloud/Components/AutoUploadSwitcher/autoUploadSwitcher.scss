.auto_upload_switcher {
  float: right;
}
