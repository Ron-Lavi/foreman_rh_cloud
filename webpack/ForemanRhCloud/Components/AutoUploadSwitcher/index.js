import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './AutoUploadSwitcherActions';
import AutoUploadSwitcher from './AutoUploadSwitcher';
import { selectAutoUploadEnabled } from '../AccountList/AccountListSelectors';

// map state to props
const mapStateToProps = state => ({
  autoUploadEnabled: selectAutoUploadEnabled(state),
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export connected component
export default connect(mapStateToProps, mapDispatchToProps)(AutoUploadSwitcher);
