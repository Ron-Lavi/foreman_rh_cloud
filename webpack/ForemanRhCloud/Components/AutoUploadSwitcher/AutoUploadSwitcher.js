import React from 'react';
import PropTypes from 'prop-types';
import { Switch, FieldLevelHelp } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';
import './autoUploadSwitcher.scss';

const AutoUploadSwitcher = ({ autoUploadEnabled, handleToggle }) => (
  <div className="auto_upload_switcher">
    <Switch
      size="mini"
      value={autoUploadEnabled}
      onChange={() => handleToggle(autoUploadEnabled)}
    />
    <FieldLevelHelp
      content={__(
        'Enable automatic upload of your host inventory to the Red Hat cloud'
      )}
    />
    <span>Allow Auto Upload</span>
  </div>
);

AutoUploadSwitcher.propTypes = {
  autoUploadEnabled: PropTypes.bool,
  handleToggle: PropTypes.func.isRequired,
};

AutoUploadSwitcher.defaultProps = {
  autoUploadEnabled: true,
};

export default AutoUploadSwitcher;
