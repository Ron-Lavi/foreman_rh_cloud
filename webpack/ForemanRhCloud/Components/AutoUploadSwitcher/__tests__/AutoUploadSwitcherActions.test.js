import { testActionSnapshotWithFixtures } from '@theforeman/test';
import API from 'foremanReact/API';
import { handleToggle } from '../AutoUploadSwitcherActions';
import { handleToggleResponse } from '../AutoUploadSwitcher.fixtures';

jest.mock('foremanReact/API');
API.post.mockImplementation(async () => handleToggleResponse);

const fixtures = {
  'should handleToggle': () => handleToggle(),
};

describe('AutoUploadSwitcher actions', () =>
  testActionSnapshotWithFixtures(fixtures));
