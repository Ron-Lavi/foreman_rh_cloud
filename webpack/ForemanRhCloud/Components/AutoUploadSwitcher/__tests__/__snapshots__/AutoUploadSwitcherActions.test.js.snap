// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoUploadSwitcher actions should handleToggle 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "autoUploadEnabled": true,
      },
      "type": "AUTO_UPLOAD_TOGGLE",
    },
  ],
]
`;
